internal/data_generator/internal/base_fields.go

```diff
@@ -0,0 +1,182 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"os"
+	"strings"
+)
+
+const accessorSliceTemplate = `// ${fieldName} returns the ${originFieldName} associated with this ${structName}.
+func (ms ${structName}) ${fieldName}() ${returnType} {
+	return new${returnType}(&ms.orig.${originFieldName})
+}
+
+// Set${fieldName} replaces the ${originFieldName} associated with this ${structName}.
+func (ms ${structName}) Set${fieldName}(v ${returnType}) {
+	ms.orig.${originFieldName} = *v.orig
+}`
+
+const accessorMessageTemplate = `// ${fieldName} returns the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) ${fieldName}() ${returnType} {
+	if rm.orig.${originFieldName} == nil {
+		// No ${originFieldName} available, initialize one to make all operations on ${returnType} available.
+		rm.orig.${originFieldName} = &${structOriginFullName}{}
+	}
+	return new${returnType}(&ms.orig.${originFieldName})
+}
+
+// Set${fieldName} replaces the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) Set${fieldName}(v ${returnType}) {
+	ms.orig.${originFieldName} = *v.orig
+}`
+
+const accessorPrimitiveTemplate = `// ${fieldName} returns the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) ${fieldName}() ${returnType} {
+	return ms.orig.Get${originFieldName}()
+}
+
+// Set${fieldName} replaces the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) Set${fieldName}(v ${returnType}) {
+	ms.orig.${originFieldName} = v
+}`
+
+const accessorPrimitiveTypedTemplate = `// ${fieldName} returns the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) ${fieldName}() ${returnType} {
+	return ${returnType}(ms.orig.Get${originFieldName}())
+}
+
+// Set${fieldName} replaces the ${lowerFieldName} associated with this ${structName}.
+func (ms ${structName}) Set${fieldName}(v ${returnType}) {
+	ms.orig.${originFieldName} = ${rawType}(v)
+}`
+
+type baseField interface {
+	generateAccessors(ms *messageStruct, sb *strings.Builder)
+}
+
+type sliceField struct {
+	fieldMame       string
+	originFieldName string
+	returnSlice     *sliceStruct
+}
+
+func (mf *sliceField) generateAccessors(ms *messageStruct, sb *strings.Builder) {
+	sb.WriteString(os.Expand(accessorSliceTemplate, func(name string) string {
+		switch name {
+		case "structName":
+			return ms.structName
+		case "fieldName":
+			return mf.fieldMame
+		case "returnType":
+			return mf.returnSlice.structName
+		case "elementStructName":
```

Comment:
nit: it doesn't break anything but I didn't see this one anywhere else in the PR nor in `accessorSliceTemplate`

Reply:
done.